interfaces/devdoc/threadpool_requirements.md

```diff
@@ -104,6 +110,60 @@ MOCKABLE_FUNCTION(, void, threadpool_close, THREADPOOL_HANDLE, threadpool);
 
 **NON_THREADPOOL_01_019: [** If `threadpool` is not OPEN, `threadpool_close` shall return. **]**
 
+### threadpool_create_work_item
+
+```c
+MOCKABLE_FUNCTION(, PVOID, threadpool_create_work_item, THANDLE(THREADPOOL), threadpool, THREADPOOL_WORK_FUNCTION, work_function, PVOID, work_function_context);
+```
+
+`threadpool_create_work_item` creates a work item to be executed by the threadpool.
+
+**NON_THREADPOOL_05_001: [** `threadpool_create_work_item` shall initialize the `work_item_context` to `NULL`, where `NULL` would be a failure indicator to the calling functions **]**
+
+**NON_THREADPOOL_05_002: [** If `threadpool` is `NULL`, `threadpool_create_work_item` shall fail and return a `NULL` value. **]**
+
+**NON_THREADPOOL_05_003: [** If `work_function` is `NULL`, `threadpool_create_work_item` shall fail and return a `NULL` value. **]**
+
+**NON_THREADPOOL_05_004: [** Otherwise `threadpool_create_work_item` shall allocate a context where `work_function`, `work_function_context`, and `ptp_work` shall be saved. **]**
+
+**NON_THREADPOOL_05_005: [** If any error occurs, `threadpool_create_work_item` shall fail and return a `NULL` value. **]**
+
+**NON_THREADPOOL_05_006: [** `threadpool_create_work_item` shall create `PTP_WORK` by calling `CreateThreadpoolWork` to set the callback function as `on_work_callback_v2` and pass the newly created context whenever the callback is called in the tp_environment. **]**
+
+**NON_THREADPOOL_05_007: [** If any error occurs, `threadpool_create_work_item` shall fail, free the newly created context and return a `NULL` value. **]**
+
+### threadpool_schedule_work_item
+
+```c
+MOCKABLE_FUNCTION(, int, threadpool_schedule_work_item, THANDLE(THREADPOOL), threadpool, PVOID, work_item_context);
+```
+
+`threadpool_schedule_work_item` schedules a work item to be executed by the threadpool.
+
+**NON_THREADPOOL_05_008: [** If `threadpool` is NULL, `threadpool_schedule_work_item` shall fail and return a `non-zero` value. **]**
+
+**NON_THREADPOOL_05_009: [** If `work_function` is NULL, `threadpool_schedule_work_item` shall fail and return a `non-zero` value. **]**
+
+**NON_THREADPOOL_05_010: [** If `ptp_work` within the context is `NULL`, `threadpool_schedule_work_item` shall fail and return a `non-zero` value. **]**
+
+**NON_THREADPOOL_05_011: [** `threadpool_schedule_work_item` shall call `SubmitThreadpoolWork` to submit the work item for execution. **]**
+
+### threadpool_work_context_destroy
+
+```c
+MOCKABLE_FUNCTION(, void, threadpool_work_context_destroy, PVOID, work_item_context);
+```
+
+`threadpool_work_context_destroy` closes the `ptp_work` member variable in `work_item_context` of type `PWORK_ITEM_CONTEXT_V2` and then frees the `work_item_context`
```

Comment:
>closes the `ptp_work` member variable in `work_item_context` of type `PWORK_ITEM_CONTEXT_V2` and then frees the `work_item_context` [](http://example.com/codeflow?start=34&length=131)

As a generic interface description this "frees the work item created by `threadpool_create_work_item`".